src/zarr/abc/store.py

```diff
@@ -221,6 +221,14 @@ def close(self) -> None:
         self._is_open = False
         pass
 
+    async def _set_dict(self, dict: Mapping[str, Buffer]) -> None:
+        """
+        Insert objects into storage as defined by a prefix: value mapping.
+        """
+        for key, value in dict.items():
+            await self.set(key, value)
```

Comment:
would prefer asyncio.gather as the default here, since each coroutine in principle may be latency bound

Reply:
switched to asyncio.gather in e4101b7a